lib/bolt/plugin/terraform.rb

```diff
@@ -49,11 +68,37 @@ def inventory_targets(opts)
       end
 
       def load_statefile(opts)
+        statefile = if opts['backend'] == 'remote'
+                      load_remote_statefile(opts)
+                    else
+                      load_local_statefile(opts)
+                    end
+
+        JSON.parse(statefile)
+      end
+
+      # Uses the Terraform CLI to pull remote state files
+      def load_remote_statefile(opts)
         dir = opts['dir']
-        filename = opts.fetch('statefile', 'terraform.tfstate')
-        statefile = File.expand_path(File.join(dir, filename))
+        cwd = Dir.pwd
+
+        Dir.chdir(dir)
+        stdout_str, stderr_str, = Open3.capture3('terraform state pull', binmode: true)
```

Comment:
Check out the `chdir` option for capture3 https://www.rubydoc.info/stdlib/core/Process.spawn . You can do something like: `Open3.capture3('terraform state pull', chdir: opts['dir'])`. Also, what is the `:binmode` option used for here?

Reply:
Updated.

I had put in `binmode` based on Glenn's use of `capture3` in the docker CLI pull: https://github.com/puppetlabs/bolt/pull/1075/files#diff-7bd8cc88b6b442a98afb7f1cef570a6aR173

I'm not sure if there would be a concern with Windows compatibility / line ending translations here without binary mode being set.